File: src/main/java/com/imcode/imcms/dao/SystemDao.java
package com.imcode.imcms.dao;

import com.imcode.imcms.api.MenuHistory;
import com.imcode.imcms.api.SystemProperty;
import imcode.server.document.textdocument.MenuDomainObject;
import imcode.server.document.textdocument.MenuItemDomainObject;
import org.hibernate.criterion.DetachedCriteria;
import org.springframework.orm.hibernate3.HibernateTemplate;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Map;

public class SystemDao extends HibernateTemplate {

	@Transactional
	public List<SystemProperty> getProperties() {
        return find("SELECT p FROM SystemProperty p");
    }


	@Transactional
	public SystemProperty getProperty(String name) {
        return (SystemProperty)getSession().createQuery("SELECT p FROM SystemProperty p WHERE p.name = ?")
                .setParameter(0, name)
                .uniqueResult();
    }
    

	@Transactional
	public void saveProperty(SystemProperty property) {
        saveOrUpdate(property);
    }
}

File: src/main/scala/com/imcode/imcms/dao/SystemDao.scala
package com.imcode
package imcms.dao

import com.imcode.imcms.mapping.orm.SystemProperty
import org.springframework.transaction.annotation.Transactional
import com.imcode.imcms.dao.hibernate.HibernateSupport


@Transactional(rollbackFor = Array(classOf[Throwable]))
class SystemDao extends HibernateSupport {

  def getProperties(): JList[SystemProperty] = hibernate.listAll()

  def getProperty(name: String): SystemProperty = hibernate.getByQuery(
    "SELECT p FROM SystemProperty p WHERE p.name = ?1", 1 -> name)

  def saveProperty(property: SystemProperty) = hibernate.saveOrUpdate(property)
}





